lcls-twincat-optics/lcls_twincat_optics_plc/POUs/FB_PitchControl.TcPOU

```diff
@@ -0,0 +1,266 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <POU Name="FB_PitchControl" Id="{143b5005-3e50-4e14-8a6a-21db919333f0}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_PitchControl
+VAR_IN_OUT
+    Pitch : HOMS_PitchMechanism;
+END_VAR
+VAR_INPUT
+END_VAR
+VAR_OUTPUT
+	q_bError : BOOL;
+	q_bDone : BOOL;
+	q_bBusy : BOOL;
+END_VAR
+VAR
+	// Logging
+	stDiag : ST_fbDiagnostics;
+	fbFormatString : FB_FormatString;
+	{attribute 'instance-path'}
+	{attribute 'no_init'}
+	POUName : STRING; // Name of the POU for logging/error reporting
+
+	// Stepper Motion
+    lrActPos : LREAL; // Actual Position of piezo mechanism
+	lrPrevStepperPos : LREAL; // Previous successfully achieved stepper position
+	ftLimitSwitch : F_TRIG;
+	ftSTO : F_TRIG;
+	rtSTO : R_TRIG;
+	lrOriginalPosRequest : LREAL; // Used for logging
+	lrLastSetpoint : LREAL; // Previous successfully achieved setpoint
+	lrCurrentSetpoint : LREAL; // Agrees with DUT_MotionStage.fPosition
+	fbMotionRequest : FB_MotionRequest;
+	fbMotionStage : FB_MotionStage;
+	bLimitHit : BOOL;
+	bSTOHit : BOOL;
+	tonStepperHold : TON := (PT:=T#100MS); // Timer to hold stepper position while the system relaxes
+	rSettledRange : REAL := 5.0; // Units = urad
+	bResetStepper : BOOL;
+	bExecuteStepper : BOOL;
+	enumMotionRequest : ENUM_MotionRequest := ENUM_MotionRequest.WAIT; // Check on this - guessing it means to wait for move to complete before requesting another
```

Comment:
The comment here is correct, WAIT is intended to be "wait for the in progress move to finish before starting a new move" when using FB_MoveRequest. The idea is that the plc code writer is the one sequencing all the (non-EPICS) moves and knows best about whether it's better to wait or to interrupt an in-progress move.

Reply:
Resolved in cd0cc6d